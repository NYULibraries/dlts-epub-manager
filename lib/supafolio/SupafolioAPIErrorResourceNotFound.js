// See https://jira.nyu.edu/jira/browse/NYUP-738 for documentation on possible
// Supafolio API errors.

const SupafolioApiError = require( './SupafolioApiError' ).SupafolioApiError;

class SupafolioApiErrorResourceNotFound extends SupafolioApiError {
    static MESSAGE = 'Resource not found.';

    constructor() {
        super(
            'SupafolioApiErrorResourceNotFound',
            SupafolioApiErrorResourceNotFound.MESSAGE,
        );
    }
}

module.exports.SupafolioApiErrorResourceNotFound = SupafolioApiErrorResourceNotFound;
