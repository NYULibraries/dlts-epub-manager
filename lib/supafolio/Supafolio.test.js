"use strict";

const fs   = require( 'fs' ),
      path = require( 'path' );

const Supafolio = require( './Supafolio' ).Supafolio;
const SupafolioAPIErrorProductNotInDatabase =
    require( './SupafolioAPIErrorProductNotInDatabase' ).SupafolioAPIErrorProductNotInDatabase;
const SupafolioAPIErrorResourceNotFound =
    require( './SupafolioAPIErrorResourceNotFound' ).SupafolioAPIErrorResourceNotFound;
const SupafolioApiStub = require( '../../test/acceptance/SupafolioApiStub' );

const LES_MISERABLES = '9780451419439';

describe( 'lib/supafolio/Supafolio.book', () => {
    let supafolio;

    beforeAll( () => {
        const SUPAFOLIO_API_KEY = require(
            path.join( __dirname, '../../test/acceptance/fixture/config-private/metadata-full.json' )
        ).supafolioApiKey;

        const supafolioApiStub = new SupafolioApiStub( SUPAFOLIO_API_KEY );

        supafolio = new Supafolio(
            SUPAFOLIO_API_KEY,
            supafolioApiStub.request.bind( supafolioApiStub ),
        );
    } );

    it( 'should return a correct Book for 9780814706404', () => {
        const book = supafolio.book( '9780814706404' );

        expect( book ).toMatchSnapshot();
    } );

    it( 'should throw a SupafolioAPIErrorResourceNotFound if given an empty ISBN', () => {
        expect( () => {
            const book = supafolio.book( '' );
        } ).toThrowError( SupafolioAPIErrorResourceNotFound );
    } );

    it( 'should throw a SupafolioAPIErrorProductNotInDatabase if ISBN is not valid', () => {
        expect( () => {
            const book = supafolio.book( '123456789' );
        } ).toThrowError( SupafolioAPIErrorProductNotInDatabase );
    } );

    it( 'should throw a SupafolioAPIErrorProductNotInDatabase if ISBN not found', () => {
        expect( () => {
            const book = supafolio.book( LES_MISERABLES );
        } ).toThrowError( SupafolioAPIErrorProductNotInDatabase );
    } );
} );

