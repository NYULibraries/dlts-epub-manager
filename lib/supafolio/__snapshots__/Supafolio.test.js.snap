// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/supafolio/Supafolio.book should return a correct Book for 9780814706404 1`] = `
Book {
  "book": Object {
    "assets": Array [
      Object {
        "asset": Object {
          "description": null,
          "embed": null,
          "height": 648,
          "id": "360",
          "name": "648-H-9780814706404.jpg",
          "order": 1,
          "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
          "schedule": "2020-01-20 10:44:25",
          "size": null,
          "source": "s3-image-import",
          "sub_type": "book-cover",
          "type": "image",
          "unique": "c6febd557ac58aedab5399d4cc207a8f",
          "width": null,
        },
        "asset_id": "360",
        "id": "360-92",
      },
      Object {
        "asset": Object {
          "description": null,
          "embed": null,
          "height": null,
          "id": "92",
          "name": "145-W-9780814706404.jpg",
          "order": 1,
          "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
          "schedule": "2020-01-20 10:44:25",
          "size": null,
          "source": "s3-image-import",
          "sub_type": "book-cover",
          "type": "image",
          "unique": "34a9f5486aa8a4ce5cd1ed1190b6497a",
          "width": 145,
        },
        "asset_id": "92",
        "id": "92-92",
      },
      Object {
        "asset": Object {
          "description": null,
          "embed": null,
          "height": null,
          "id": "226",
          "name": "298-W-9780814706404.jpg",
          "order": 1,
          "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
          "schedule": "2020-01-20 10:44:25",
          "size": null,
          "source": "s3-image-import",
          "sub_type": "book-cover",
          "type": "image",
          "unique": "fe5b3688cfd4056ba97ed8955cc152b6",
          "width": 298,
        },
        "asset_id": "226",
        "id": "226-92",
      },
    ],
    "availability": 4,
    "availability_status": "Active",
    "awards": Array [],
    "categories": Array [
      Object {
        "category": Object {
          "code": "NYUAFS000",
          "code_3": "NYU",
          "code_6": "NYUAFS",
          "name": "African American Studies",
          "seo_name": "NYUAFS",
          "type": "nyuos-subjects",
        },
        "primary": false,
      },
      Object {
        "category": Object {
          "code": "NYUL00000",
          "code_3": "NYU",
          "code_6": "NYUL00",
          "name": "Law",
          "seo_name": "NYUL",
          "type": "nyuos-subjects",
        },
        "primary": false,
      },
      Object {
        "category": Object {
          "code": "NYUS00000",
          "code_3": "NYU",
          "code_6": "NYUS00",
          "name": "Sociology",
          "seo_name": "NYUS",
          "type": "nyuos-subjects",
        },
        "primary": false,
      },
    ],
    "cd_run_time": "0:0:0",
    "contributors": Array [
      Object {
        "contributor": Object {
          "bio": "<b>Jody David Armour </b>is Professor of Law at the School of Law, University of Southern California.",
          "bio_short": null,
          "contributor_id": 100029,
          "custom": null,
          "deceased": false,
          "featured": false,
          "first": "Jody David",
          "hide": false,
          "image": null,
          "last": "Armour",
          "locale": null,
          "name": "Jody David Armour",
          "on_tour": false,
          "reference_id": null,
          "role": "A01",
          "sensitivity": false,
          "seo": "jody-david-armour",
        },
        "contributor_id": 100029,
        "note": "Jody David Armour",
        "order": 1,
        "role": Object {
          "code": "A01",
          "name": "By (author)",
        },
        "role_id": "A01",
      },
    ],
    "custom": Object {
      "cs_title_group": "9780814706404",
      "ebook": "9780814707494",
      "edition_isbn": "9780814706404",
      "hardcover": "9780814706404",
      "hi_res_images": false,
      "language": "eng",
      "library": "9780814707821",
      "license_link": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
      "license_name": "Creative Commons Attribution-NonCommercial-ShareAlike-4.0-International-Public-License",
      "open_access_is_downloadable": false,
      "open_access_series": false,
      "paperback": "9780814706701",
      "permanent_url": "https://doi.org/10.33682/37pvmfhh",
      "scheduled_date": "05/29/2014",
      "video": false,
    },
    "date": Object {
      "__className": "DateTime",
      "date": "1997-05-01 00:00:00.000000",
      "timezone": "America/New_York",
      "timezone_type": 3,
    },
    "description": "<p>Tackling the ugly secret of unconscious racism in American society, this book provides specific solutions to counter this entrenched phenomenon.</p>",
    "division": null,
    "edition": "",
    "featured": false,
    "formats": Array [
      Object {
        "date": Object {
          "__className": "DateTime",
          "date": "1997-05-01 00:00:00.000000",
          "timezone": "America/New_York",
          "timezone_type": 3,
        },
        "detail": "Print PDF",
        "edition": "",
        "format": Object {
          "code": "BB",
          "name": "Hardcover",
        },
        "format_id": "BB",
        "id": "BB92",
        "isbn": "9780814706404",
        "primary_format": false,
        "product_id": 92,
        "seo_link": "negrophobia-and-reasonable-racism",
        "title_id": "512b132534fc22435138230edbf4f359",
      },
      Object {
        "date": Object {
          "__className": "DateTime",
          "date": "1997-05-01 00:00:00.000000",
          "timezone": "America/New_York",
          "timezone_type": 3,
        },
        "detail": "",
        "edition": "",
        "format": Object {
          "code": "DG",
          "name": "eBook",
        },
        "format_id": "DG",
        "id": "DG24",
        "isbn": "9780814707821",
        "primary_format": false,
        "product_id": 24,
        "seo_link": "negrophobia-and-reasonable-racism",
        "title_id": "512b132534fc22435138230edbf4f359",
      },
      Object {
        "date": Object {
          "__className": "DateTime",
          "date": "1997-05-01 00:00:00.000000",
          "timezone": "America/New_York",
          "timezone_type": 3,
        },
        "detail": "",
        "edition": "",
        "format": Object {
          "code": "DG",
          "name": "eBook",
        },
        "format_id": "DG",
        "id": "DG346",
        "isbn": "9780814707494",
        "primary_format": false,
        "product_id": 346,
        "seo_link": "negrophobia-and-reasonable-racism",
        "title_id": "512b132534fc22435138230edbf4f359",
      },
      Object {
        "date": Object {
          "__className": "DateTime",
          "date": "2000-01-01 00:00:00.000000",
          "timezone": "America/New_York",
          "timezone_type": 3,
        },
        "detail": "Print PDF",
        "edition": "",
        "format": Object {
          "code": "BC",
          "name": "Paperback",
        },
        "format_id": "BC",
        "id": "BC347",
        "isbn": "9780814706701",
        "primary_format": true,
        "product_id": 347,
        "seo_link": "negrophobia-and-reasonable-racism",
        "title_id": "512b132534fc22435138230edbf4f359",
      },
    ],
    "hide": 0,
    "illustrations_note": "",
    "image": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
    "import_notes": "",
    "imprint": Object {
      "description": null,
      "description_short": null,
      "hide": false,
      "name": "NYU Press",
      "seo_name": "nyu-press",
    },
    "initial_print_run": null,
    "isbn10": "0814706401",
    "isbn13": "9780814706404",
    "keywords": "",
    "levels": Array [],
    "pages": 216,
    "prices": Array [
      Object {
        "amount": "74.00",
        "discount": null,
        "locale": "GBP",
        "promo_start": null,
        "promo_stop": null,
        "saving": null,
      },
      Object {
        "amount": "89.00",
        "discount": null,
        "locale": "USD",
        "promo_start": Object {
          "__className": "DateTime",
          "date": "2019-03-31 00:00:00.000000",
          "timezone": "America/New_York",
          "timezone_type": 3,
        },
        "promo_stop": null,
        "saving": null,
      },
      Object {
        "amount": "89.00",
        "discount": null,
        "locale": "AUD",
        "promo_start": null,
        "promo_stop": null,
        "saving": null,
      },
    ],
    "primary_format": false,
    "product_id": 92,
    "publisher": Object {
      "name": "NYU Press",
      "seo_name": "nyu-press",
    },
    "rating": null,
    "reference_id": null,
    "retailers": null,
    "reviews": Array [
      Object {
        "id": "92-6694",
        "review": Object {
          "assets": Array [],
          "description": "\\"Skillfully drawing on a wide range of referents from Greek mythology to Thomas Byes, the father of statistics, Armour plumbs our racial psychology and in the process explores the racialized nature of our daily life and our legal system.\\"",
          "id": "6694",
          "reviewer": "Norfolk Virginia Journal and Guide",
          "score": null,
          "source": "Onix",
          "unique": "9780814706404-12acbf0048eb5ee105f4abeffaea1017",
        },
        "review_id": "6694",
      },
      Object {
        "id": "92-6695",
        "review": Object {
          "assets": Array [],
          "description": "\\"A crucial book, at a critical time, just as white America is gradually coming to understand the hidden travails of African American life.\\"",
          "id": "6695",
          "reviewer": "Bookwatch",
          "score": null,
          "source": "Onix",
          "unique": "9780814706404-d3cc4ed45dd0db458000bdd580c88ef0",
        },
        "review_id": "6695",
      },
      Object {
        "id": "92-6696",
        "review": Object {
          "assets": Array [],
          "description": "\\"[C]ommendable in important respects. [Armour] points out problems in the theory of rational racial discrimination that many of its proponents ignore or overlook.\\"",
          "id": "6696",
          "reviewer": "Randall Kennedy, Los Angeles Times Book Review",
          "score": null,
          "source": "Onix",
          "unique": "9780814706404-cf03377ee5b763622a9295c2d0eebe2d",
        },
        "review_id": "6696",
      },
      Object {
        "id": "92-6697",
        "review": Object {
          "assets": Array [],
          "description": "\\"A patient meditation.\\"",
          "id": "6697",
          "reviewer": "The Nation",
          "score": null,
          "source": "Onix",
          "unique": "9780814706404-4903c6e6ab813192aeaf34f246663680",
        },
        "review_id": "6697",
      },
    ],
    "sale_date": Object {
      "__className": "DateTime",
      "date": "1997-05-01 00:00:00.000000",
      "timezone": "America/New_York",
      "timezone_type": 3,
    },
    "scheduled": Array [],
    "search_weight": 570,
    "season": null,
    "seo": "negrophobia-and-reasonable-racism",
    "series": Array [
      Object {
        "number_in_series": "32",
        "series": Object {
          "description": "",
          "name": "Critical America",
          "seo_name": "critical-america",
        },
      },
    ],
    "sort_date": 862459200,
    "stock": null,
    "subtitle": "The Hidden Costs of Being Black in America",
    "summary": "Tackling the ugly secret of unconscious racism in American society, this book provides specific solutions to counter this entrenched phenomenon.",
    "tagline": null,
    "territories": null,
    "title": "Negrophobia and Reasonable Racism",
    "title_id": "512b132534fc22435138230edbf4f359",
    "title_without_prefix": "Negrophobia and Reasonable Racism",
    "trim": Object {
      "depth": null,
      "height": "8.000",
      "unit": "in",
      "weight": "0.050",
      "weight_unit": "lb",
      "width": "5.000",
    },
    "type": "Nonfiction",
  },
  "supafolioApiResponse": Object {
    "data": Object {
      "book": Array [
        Object {
          "assets": Array [
            Object {
              "asset": Object {
                "description": null,
                "embed": null,
                "height": 648,
                "id": "360",
                "name": "648-H-9780814706404.jpg",
                "order": 1,
                "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
                "schedule": "2020-01-20 10:44:25",
                "size": null,
                "source": "s3-image-import",
                "sub_type": "book-cover",
                "type": "image",
                "unique": "c6febd557ac58aedab5399d4cc207a8f",
                "width": null,
              },
              "asset_id": "360",
              "id": "360-92",
            },
            Object {
              "asset": Object {
                "description": null,
                "embed": null,
                "height": null,
                "id": "92",
                "name": "145-W-9780814706404.jpg",
                "order": 1,
                "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
                "schedule": "2020-01-20 10:44:25",
                "size": null,
                "source": "s3-image-import",
                "sub_type": "book-cover",
                "type": "image",
                "unique": "34a9f5486aa8a4ce5cd1ed1190b6497a",
                "width": 145,
              },
              "asset_id": "92",
              "id": "92-92",
            },
            Object {
              "asset": Object {
                "description": null,
                "embed": null,
                "height": null,
                "id": "226",
                "name": "298-W-9780814706404.jpg",
                "order": 1,
                "path": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
                "schedule": "2020-01-20 10:44:25",
                "size": null,
                "source": "s3-image-import",
                "sub_type": "book-cover",
                "type": "image",
                "unique": "fe5b3688cfd4056ba97ed8955cc152b6",
                "width": 298,
              },
              "asset_id": "226",
              "id": "226-92",
            },
          ],
          "availability": 4,
          "availability_status": "Active",
          "awards": Array [],
          "categories": Array [
            Object {
              "category": Object {
                "code": "NYUAFS000",
                "code_3": "NYU",
                "code_6": "NYUAFS",
                "name": "African American Studies",
                "seo_name": "NYUAFS",
                "type": "nyuos-subjects",
              },
              "primary": false,
            },
            Object {
              "category": Object {
                "code": "NYUL00000",
                "code_3": "NYU",
                "code_6": "NYUL00",
                "name": "Law",
                "seo_name": "NYUL",
                "type": "nyuos-subjects",
              },
              "primary": false,
            },
            Object {
              "category": Object {
                "code": "NYUS00000",
                "code_3": "NYU",
                "code_6": "NYUS00",
                "name": "Sociology",
                "seo_name": "NYUS",
                "type": "nyuos-subjects",
              },
              "primary": false,
            },
          ],
          "cd_run_time": "0:0:0",
          "contributors": Array [
            Object {
              "contributor": Object {
                "bio": "<b>Jody David Armour </b>is Professor of Law at the School of Law, University of Southern California.",
                "bio_short": null,
                "contributor_id": 100029,
                "custom": null,
                "deceased": false,
                "featured": false,
                "first": "Jody David",
                "hide": false,
                "image": null,
                "last": "Armour",
                "locale": null,
                "name": "Jody David Armour",
                "on_tour": false,
                "reference_id": null,
                "role": "A01",
                "sensitivity": false,
                "seo": "jody-david-armour",
              },
              "contributor_id": 100029,
              "note": "Jody David Armour",
              "order": 1,
              "role": Object {
                "code": "A01",
                "name": "By (author)",
              },
              "role_id": "A01",
            },
          ],
          "custom": Object {
            "cs_title_group": "9780814706404",
            "ebook": "9780814707494",
            "edition_isbn": "9780814706404",
            "hardcover": "9780814706404",
            "hi_res_images": false,
            "language": "eng",
            "library": "9780814707821",
            "license_link": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
            "license_name": "Creative Commons Attribution-NonCommercial-ShareAlike-4.0-International-Public-License",
            "open_access_is_downloadable": false,
            "open_access_series": false,
            "paperback": "9780814706701",
            "permanent_url": "https://doi.org/10.33682/37pvmfhh",
            "scheduled_date": "05/29/2014",
            "video": false,
          },
          "date": Object {
            "__className": "DateTime",
            "date": "1997-05-01 00:00:00.000000",
            "timezone": "America/New_York",
            "timezone_type": 3,
          },
          "description": "<p>Tackling the ugly secret of unconscious racism in American society, this book provides specific solutions to counter this entrenched phenomenon.</p>",
          "division": null,
          "edition": "",
          "featured": false,
          "formats": Array [
            Object {
              "date": Object {
                "__className": "DateTime",
                "date": "1997-05-01 00:00:00.000000",
                "timezone": "America/New_York",
                "timezone_type": 3,
              },
              "detail": "Print PDF",
              "edition": "",
              "format": Object {
                "code": "BB",
                "name": "Hardcover",
              },
              "format_id": "BB",
              "id": "BB92",
              "isbn": "9780814706404",
              "primary_format": false,
              "product_id": 92,
              "seo_link": "negrophobia-and-reasonable-racism",
              "title_id": "512b132534fc22435138230edbf4f359",
            },
            Object {
              "date": Object {
                "__className": "DateTime",
                "date": "1997-05-01 00:00:00.000000",
                "timezone": "America/New_York",
                "timezone_type": 3,
              },
              "detail": "",
              "edition": "",
              "format": Object {
                "code": "DG",
                "name": "eBook",
              },
              "format_id": "DG",
              "id": "DG24",
              "isbn": "9780814707821",
              "primary_format": false,
              "product_id": 24,
              "seo_link": "negrophobia-and-reasonable-racism",
              "title_id": "512b132534fc22435138230edbf4f359",
            },
            Object {
              "date": Object {
                "__className": "DateTime",
                "date": "1997-05-01 00:00:00.000000",
                "timezone": "America/New_York",
                "timezone_type": 3,
              },
              "detail": "",
              "edition": "",
              "format": Object {
                "code": "DG",
                "name": "eBook",
              },
              "format_id": "DG",
              "id": "DG346",
              "isbn": "9780814707494",
              "primary_format": false,
              "product_id": 346,
              "seo_link": "negrophobia-and-reasonable-racism",
              "title_id": "512b132534fc22435138230edbf4f359",
            },
            Object {
              "date": Object {
                "__className": "DateTime",
                "date": "2000-01-01 00:00:00.000000",
                "timezone": "America/New_York",
                "timezone_type": 3,
              },
              "detail": "Print PDF",
              "edition": "",
              "format": Object {
                "code": "BC",
                "name": "Paperback",
              },
              "format_id": "BC",
              "id": "BC347",
              "isbn": "9780814706701",
              "primary_format": true,
              "product_id": 347,
              "seo_link": "negrophobia-and-reasonable-racism",
              "title_id": "512b132534fc22435138230edbf4f359",
            },
          ],
          "hide": 0,
          "illustrations_note": "",
          "image": "https://nyu-opensquare.imgix.net/missing-cover.jpg",
          "import_notes": "",
          "imprint": Object {
            "description": null,
            "description_short": null,
            "hide": false,
            "name": "NYU Press",
            "seo_name": "nyu-press",
          },
          "initial_print_run": null,
          "isbn10": "0814706401",
          "isbn13": "9780814706404",
          "keywords": "",
          "levels": Array [],
          "pages": 216,
          "prices": Array [
            Object {
              "amount": "74.00",
              "discount": null,
              "locale": "GBP",
              "promo_start": null,
              "promo_stop": null,
              "saving": null,
            },
            Object {
              "amount": "89.00",
              "discount": null,
              "locale": "USD",
              "promo_start": Object {
                "__className": "DateTime",
                "date": "2019-03-31 00:00:00.000000",
                "timezone": "America/New_York",
                "timezone_type": 3,
              },
              "promo_stop": null,
              "saving": null,
            },
            Object {
              "amount": "89.00",
              "discount": null,
              "locale": "AUD",
              "promo_start": null,
              "promo_stop": null,
              "saving": null,
            },
          ],
          "primary_format": false,
          "product_id": 92,
          "publisher": Object {
            "name": "NYU Press",
            "seo_name": "nyu-press",
          },
          "rating": null,
          "reference_id": null,
          "retailers": null,
          "reviews": Array [
            Object {
              "id": "92-6694",
              "review": Object {
                "assets": Array [],
                "description": "\\"Skillfully drawing on a wide range of referents from Greek mythology to Thomas Byes, the father of statistics, Armour plumbs our racial psychology and in the process explores the racialized nature of our daily life and our legal system.\\"",
                "id": "6694",
                "reviewer": "Norfolk Virginia Journal and Guide",
                "score": null,
                "source": "Onix",
                "unique": "9780814706404-12acbf0048eb5ee105f4abeffaea1017",
              },
              "review_id": "6694",
            },
            Object {
              "id": "92-6695",
              "review": Object {
                "assets": Array [],
                "description": "\\"A crucial book, at a critical time, just as white America is gradually coming to understand the hidden travails of African American life.\\"",
                "id": "6695",
                "reviewer": "Bookwatch",
                "score": null,
                "source": "Onix",
                "unique": "9780814706404-d3cc4ed45dd0db458000bdd580c88ef0",
              },
              "review_id": "6695",
            },
            Object {
              "id": "92-6696",
              "review": Object {
                "assets": Array [],
                "description": "\\"[C]ommendable in important respects. [Armour] points out problems in the theory of rational racial discrimination that many of its proponents ignore or overlook.\\"",
                "id": "6696",
                "reviewer": "Randall Kennedy, Los Angeles Times Book Review",
                "score": null,
                "source": "Onix",
                "unique": "9780814706404-cf03377ee5b763622a9295c2d0eebe2d",
              },
              "review_id": "6696",
            },
            Object {
              "id": "92-6697",
              "review": Object {
                "assets": Array [],
                "description": "\\"A patient meditation.\\"",
                "id": "6697",
                "reviewer": "The Nation",
                "score": null,
                "source": "Onix",
                "unique": "9780814706404-4903c6e6ab813192aeaf34f246663680",
              },
              "review_id": "6697",
            },
          ],
          "sale_date": Object {
            "__className": "DateTime",
            "date": "1997-05-01 00:00:00.000000",
            "timezone": "America/New_York",
            "timezone_type": 3,
          },
          "scheduled": Array [],
          "search_weight": 570,
          "season": null,
          "seo": "negrophobia-and-reasonable-racism",
          "series": Array [
            Object {
              "number_in_series": "32",
              "series": Object {
                "description": "",
                "name": "Critical America",
                "seo_name": "critical-america",
              },
            },
          ],
          "sort_date": 862459200,
          "stock": null,
          "subtitle": "The Hidden Costs of Being Black in America",
          "summary": "Tackling the ugly secret of unconscious racism in American society, this book provides specific solutions to counter this entrenched phenomenon.",
          "tagline": null,
          "territories": null,
          "title": "Negrophobia and Reasonable Racism",
          "title_id": "512b132534fc22435138230edbf4f359",
          "title_without_prefix": "Negrophobia and Reasonable Racism",
          "trim": Object {
            "depth": null,
            "height": "8.000",
            "unit": "in",
            "weight": "0.050",
            "weight_unit": "lb",
            "width": "5.000",
          },
          "type": "Nonfiction",
        },
      ],
      "catalogId": 10150,
    },
    "status": "success",
  },
}
`;
